File: 4th.js
class GoodDog {
    constructor(name) {
      this._name = name
    };
  
    get get_name() {
      return this._name
    };
  
    set set_name(name) {
      this._name = name
    };
  
    get speak() {
      return `${this._name} says Whoop!`
    }
};
  
  let sparky = new GoodDog("Sparky");
  console.log(sparky.get_name);
  console.log(sparky.speak);
  console.log("--- --- ---")
  let fido = new GoodDog("Fido")
  console.log(fido.set_name = "Fido")
  // the @name changed ! 
  console.log(fido.speak)

File: 4th.rb
class GoodDog
    def initialize(name)
      @name = name
    end
  
    def get_name
      @name
    end
  
    def set_name=(name) # (name) can be anything 
      @name = name # name can be anything 
    end
  
    def speak
      "#{@name} says Whoop!"
    end
  end
  
  sparky = GoodDog.new("Sparky")
  sparky.get_name
  sparky.speak
  fido = GoodDog.new("Fido")
  sparky.set_name = "fido"
  fido.speak
  